### PR TITLE
PR for SRVLOGIC-369: Add a new procedure for kn-workflow plugin using the binary download. 

### PR DESCRIPTION
**Pull request by kaldesai**
**Affecting version:** `serverless-docs-1.33`, `serverless-docs-1.34`

**Tracking JIRA:** https://issues.redhat.com/browse/SRVLOGIC-369

**Doc previews:** 

- [Installing the OpenShift Serverless Logic Knative Workflow plugin for Linux](https://80122--ocpdocs-pr.netlify.app/openshift-serverless/latest/install/serverless-logic-install-kn-workflow-plugin-cli.html#serverless-logic-install-kn-workflow-binary-file-linux_serverless-logic-install-kn-workflow-artifact-images)
- [Installing the OpenShift Serverless Logic Knative Workflow plugin for macOS](https://80122--ocpdocs-pr.netlify.app/openshift-serverless/latest/install/serverless-logic-install-kn-workflow-plugin-cli.html#serverless-logic-install-kn-workflow-binary-file-macos_serverless-logic-install-kn-workflow-artifact-images)
- [Installing the OpenShift Serverless Logic Knative Workflow plugin for Windows](https://80122--ocpdocs-pr.netlify.app/openshift-serverless/latest/install/serverless-logic-install-kn-workflow-plugin-cli.html#serverless-logic-install-kn-workflow-binary-file-windows_serverless-logic-install-kn-workflow-artifact-images)

### CODE REVIEW

File: modules/serverless-logic-install-kn-workflow-binary-file.adoc [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+
+Available Commands:
+  completion  Generate the autocompletion script for the specified shell
+  create      Creates a new OpenShift Serverless Logic Workflow project
+  deploy      Deploy an OpenShift Serverless Logic Workflow project on Kubernetes via SonataFlow Operator
+  help        Help about any command
+  quarkus     Manage OpenShift Serverless Logic Workflow projects built in Quarkus
+  run         Run an OpenShift Serverless Logic Workflow project in development mode
+  undeploy    Undeploy an OpenShift Serverless Logic Workflow project on Kubernetes via SonataFlow Operator
+  version     Show the version
```
Comment (domhanak) [inline modules/serverless-logic-install-kn-workflow-binary-file.adoc:83]: this output looks outdated

Comment (kaldesai) [reply modules/serverless-logic-install-kn-workflow-binary-file.adoc:83]: Do you have the latest one that I can include?

File: modules/serverless-logic-install-kn-workflow-binary-file.adoc [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+= Installing the OpenShift Serverless Logic Knative Workflow plugin using the binary file
+
+.Prerequisites
+
+* You have installed the Knative (`kn`) CLI.
+* You have installed Podman on your local machine. 
+
+.Procedure
+
+. Download the latest binary file, suitable for your environment, from the link:https://mirror.openshift.com/pub/cgw/serverless-logic/latest/[Serverless Logic download mirror] page.
```
Comment (domhanak) [inline modules/serverless-logic-install-kn-workflow-binary-file.adoc:16]: I suggest we split it per OS, same as KNative parts.

image

File: modules/serverless-logic-install-kn-workflow-binary-file.adoc [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+[source,terminal]
+----
+$ tar xvzf kn-workflow-linux-amd64.tar.gz
+----
+
+. Rename the Knative Workflow plugin binary file to `kn-workflow` by running the following command:
++
+[source,terminal]
+----
+$ mv kn kn-workflow
```
Comment (domhanak) [inline modules/serverless-logic-install-kn-workflow-binary-file.adoc:29]: this does not correspond to result of `tar xvzf kn-workflow-linux-amd64.tar.gz`. Running this I get: `kn-workflow-linux-amd64` not `kn`

File: modules/serverless-logic-install-kn-workflow-binary-file.adoc [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+----
+
+. Install the `kn-workflow` command as a plugin of the Knative CLI by running the following commands:
```
Comment (domhanak) [inline modules/serverless-logic-install-kn-workflow-binary-file.adoc:32]:
```suggestion
. Install the `workflow` command as a plugin of the Knative CLI by running the following commands:
```

File: modules/serverless-logic-install-kn-workflow-binary-file.adoc [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+
+.. Copy the `kn-workflow` binary file to a directory in your PATH, such as `/usr/local/bin`, and ensure the file name is `kn-workflow`:
```
Comment (domhanak) [inline modules/serverless-logic-install-kn-workflow-binary-file.adoc:34]:
```suggestion
.. Copy the `kn-workflow` binary file to a directory in your PATH, such as `/usr/local/bin`:
```

Removing this as there is step the ensures this.

File: modules/serverless-logic-install-kn-workflow-binary-file.adoc [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
++
+[source,terminal]
+----
+$ cp path/to/downloaded/kn-workflow /usr/local/bin/kn-workflow
+----
+
+.. Make the binary file executable:
```
Comment (domhanak) [inline modules/serverless-logic-install-kn-workflow-binary-file.adoc:41]: Imho let's use the same order we have in midstream, see https://openshift-knative.github.io/docs/docs/latest/serverless-logic/cloud/operator/install-kn-workflow-cli.html

1. Rename
2. Make executabel
3. Copy to location in PATH

File: modules/serverless-logic-install-kn-workflow-binary-file.adoc [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+----
+$ kn plugin list
+----
+
+. After installing the plugin, you can use `kn-workflow` to run the related subcommands.
++
+.Aliases to use workflow subcommand
+[source,terminal]
+----
+kn-workflow
```
Comment (domhanak) [inline modules/serverless-logic-install-kn-workflow-binary-file.adoc:61]:
```suggestion
kn workflow
kn-workflow
```

**Review by ricardozanini — APPROVED** [review]
LGTM as soon as @domhanak comments are fixed.

**Comment by kaldesai** [comment]
@domhanak, per your suggestions I have created different sections for Linux, macOS, and Windows. Can you please verify?

**Review by domhanak — APPROVED** [review]

File: modules/serverless-logic-install-kn-workflow-binary-file-macos.adoc [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+----
+
+. Rename the Knative Workflow plugin binary file to `kn-workflow` by running the following command:
++
+[source,terminal]
+----
+$ mv kn-workflow-darwin-amd64.exe kn-workflow
+----
+
+. Install the `kn-workflow` command as a plugin of the Knative CLI by running the following commands:
```
Comment (xenolinux) [inline modules/serverless-logic-install-kn-workflow-binary-file-macos.adoc:38]:
```suggestion
. Install the `kn-workflow` command as the Knative CLI plugin. 
```

File: modules/serverless-logic-install-kn-workflow-binary-file-macos.adoc [hunk 9]
```diff
@@ -0,0 +1,60 @@
...
+
+.. Make the binary file executable:
```
Comment (xenolinux) [inline modules/serverless-logic-install-kn-workflow-binary-file-macos.adoc:40]:
```suggestion
.. Make the binary file executable by running the following command:
```

File: modules/serverless-logic-install-kn-workflow-binary-file-macos.adoc [hunk 10]
```diff
@@ -0,0 +1,60 @@
...
++
+[source,terminal]
+----
+$ chmod +x /usr/local/bin/kn-workflow
+----
+
+.. Copy the `kn-workflow` binary file to a directory in your PATH, such as `/usr/local/bin`:
```
Comment (xenolinux) [inline modules/serverless-logic-install-kn-workflow-binary-file-macos.adoc:47]:
```suggestion
.. Copy the `kn-workflow` binary file to a directory in the `/usr/local/bin` path by running the following command:
```

File: modules/serverless-logic-install-kn-workflow-binary-file-macos.adoc [hunk 11]
```diff
@@ -0,0 +1,60 @@
...
++
+[source,terminal]
+----
+$ cp path/to/downloaded/kn-workflow /usr/local/bin/kn-workflow
+----
+
+.Verification
+. Run the following command to verify that the `kn-workflow` plugin is installed successfully:
```
Comment (xenolinux) [inline modules/serverless-logic-install-kn-workflow-binary-file-macos.adoc:55]:
```suggestion
.Verification

* Run the following command to verify that the `kn-workflow` plugin is installed successfully:
```

https://redhat-documentation.github.io/supplementary-style-guide/#single-step-procedures

File: modules/serverless-logic-install-kn-workflow-binary-file-windows.adoc [hunk 12]
```diff
@@ -0,0 +1,50 @@
...
+
+If you are using Windows, you can install the Knative Workflow plugin as a binary file. To do this, you must download and unpack a ZIP archive and add the binary to a directory on your `PATH`.
+
+.Prerequisites
+
+* You have installed the Knative (`kn`) CLI.
+
+.Procedure
+
+. Download the latest `ZIP` archive suitable for your environment, from the link:https://mirror.openshift.com/pub/cgw/serverless-logic/latest/[Serverless Logic download mirror] page.
```
Comment (xenolinux) [inline modules/serverless-logic-install-kn-workflow-binary-file-windows.adoc:17]:
```suggestion
. Download the latest `zip` archive suitable for your environment, from the link:https://mirror.openshift.com/pub/cgw/serverless-logic/latest/[Serverless Logic download mirror] page.
```

File: modules/serverless-logic-install-kn-workflow-binary-file-windows.adoc [hunk 13]
```diff
@@ -0,0 +1,50 @@
...
+
+. Extract the selected Knative Workflow plugin binary file using PowerShell by running the following command:
```
Comment (xenolinux) [inline modules/serverless-logic-install-kn-workflow-binary-file-windows.adoc:19]:
```suggestion
. Extract the Knative Workflow plugin binary file by using PowerShell by running the following command:
```

File: modules/serverless-logic-install-kn-workflow-binary-file-windows.adoc [hunk 14]
```diff
@@ -0,0 +1,50 @@
...
++
+[source,terminal]
+----
+$ Expand-Archive -Path <filename>.zip -DestinationPath <destination>
+----
+
+. Rename the Knative Workflow plugin binary file to `kn-workflow` by running the following command:
```
Comment (xenolinux) [inline modules/serverless-logic-install-kn-workflow-binary-file-windows.adoc:26]:
```suggestion
. Rename the extracted Knative Workflow plugin binary file to `kn-workflow` by running the following command:
```

File: modules/serverless-logic-install-kn-workflow-binary-file-windows.adoc [hunk 15]
```diff
@@ -0,0 +1,50 @@
...
++
+[source,terminal]
+----
+$ Rename-Item -Path <destination>\<filename>.exe -NewName kn-workflow.exe
+----
+
+. Install the `kn-workflow` command as a plugin of the Knative CLI by running the following commands:
```
Comment (xenolinux) [inline modules/serverless-logic-install-kn-workflow-binary-file-windows.adoc:33]:
```suggestion
. Install the `kn-workflow` command as the Knative CLI plugin. 
```

File: modules/serverless-logic-install-kn-workflow-binary-file-windows.adoc [hunk 16]
```diff
@@ -0,0 +1,50 @@
...
+
+.. Make the binary file executable.
```
Comment (xenolinux) [inline modules/serverless-logic-install-kn-workflow-binary-file-windows.adoc:35]:
```suggestion
.. Make the binary file executable by running the following command:
```

File: modules/serverless-logic-install-kn-workflow-binary-file-windows.adoc [hunk 17]
```diff
@@ -0,0 +1,50 @@
+// Module included in the following assemblies:
+//
+// * /serverless/install/serverless-logic-install-kn-workflow-plugin-cli.adoc
+
+:_content-type: PROCEDURE
+[id="serverless-logic-install-kn-workflow-binary-file-windows_{context}"]
+= Installing the OpenShift Serverless Logic Knative Workflow plugin for Windows
+
+If you are using Windows, you can install the Knative Workflow plugin as a binary file. To do this, you must download and unpack a ZIP archive and add the binary to a directory on your `PATH`.
```
Comment (xenolinux) [inline modules/serverless-logic-install-kn-workflow-binary-file-windows.adoc:9]:
```suggestion
If you are using Windows, you can install the Knative Workflow plugin.
```
Same comment as Linux and macos modules.

File: modules/serverless-logic-install-kn-workflow-binary-file-windows.adoc [hunk 18]
```diff
@@ -0,0 +1,50 @@
...
+
+.Prerequisites
+
+* You have installed the Knative (`kn`) CLI.
```
Comment (xenolinux) [inline modules/serverless-logic-install-kn-workflow-binary-file-windows.adoc:13]:
```suggestion
* You have installed the Knative (`kn`) command-line interface (CLI).
```

File: modules/serverless-logic-install-kn-workflow-binary-file-windows.adoc [hunk 19]
```diff
@@ -0,0 +1,50 @@
...
+[source,terminal]
+----
+$ Rename-Item -Path <destination>\<filename>.exe -NewName kn-workflow.exe
+----
+
+. Install the `kn-workflow` command as a plugin of the Knative CLI by running the following commands:
+
+.. Make the binary file executable.
+
+.. Copy the `kn-workflow` binary file to a directory in your PATH by running the following command:
```
[the remark this answers is not in the document]
Comment (xenolinux) [reply modules/serverless-logic-install-kn-workflow-binary-file-windows.adoc:37]: I am not sure if `PATH` -- the environment variable should be used. Should it be "path"?

**Review by xenolinux — COMMENTED** [review]
Left some suggestions and comments. Nice work! 

Please let me know if you have questions. Thanks

/remove-label peer-review-needed
/remove-label peer-review-in-progress
/label peer-review-done

File: modules/serverless-logic-install-kn-workflow-binary-file-macos.adoc [hunk 20]
```diff
@@ -0,0 +1,60 @@
+// Module included in the following assemblies:
+//
+// * /serverless/install/serverless-logic-install-kn-workflow-plugin-cli.adoc
+
+:_content-type: PROCEDURE
+[id="serverless-logic-install-kn-workflow-binary-file-macos_{context}"]
+= Installing the OpenShift Serverless Logic Knative Workflow plugin for macOS
```
Comment (xenolinux) [inline modules/serverless-logic-install-kn-workflow-binary-file-macos.adoc:7]: Same comment as https://github.com/openshift/openshift-docs/pull/80122#discussion_r1708774397

File: modules/serverless-logic-install-kn-workflow-binary-file-macos.adoc [hunk 21]
```diff
@@ -0,0 +1,60 @@
...
+
+If you are using macOS, you can install the Knative Workflow plugin as a binary file. To do this, you must download and unpack a `tar.gz` archive and add the binary to a directory on your `PATH`.
+
+[IMPORTANT]
+====
+On macOS, some systems might block the application from running due to Apple's security policies. To fix this issue, check the *Security & Privacy* section in the *System Preferences* under the *General* tab to approve the application to run. For more information, see the link:https://support.apple.com/en-in/guide/mac-help/mh40616/mac[Open a Mac app from an unidentified developer] Apple support article.
```
[the remark this answers is not in the document]
Comment (kaldesai) [reply modules/serverless-logic-install-kn-workflow-binary-file-macos.adoc:13]: Okay, noted!

File: modules/serverless-logic-install-kn-workflow-binary-file-windows.adoc [hunk 22]
```diff
@@ -0,0 +1,57 @@
...
+
+. Install the `kn-workflow` command as the Knative CLI plugin.
++
+On Windows, files like `exe` are treated as executable by default, so you do not need to change permissions.
+
+.. Move the `kn-workflow.exe` file to a directory in your `PATH` by running the following command:
++
+[source,terminal]
+----
+$ move C:\path\to\kn-workflow.exe C:\Program Files\
```
Comment (aireilly) [inline modules/serverless-logic-install-kn-workflow-binary-file-windows.adoc:41]:
```suggestion
$ move C:\path\to\kn-workflow.exe C:\Program Files
```

File: modules/serverless-logic-install-kn-workflow-binary-file-linux.adoc [hunk 23]
```diff
@@ -0,0 +1,67 @@
+// Module included in the following assemblies:
+//
+// * /serverless/install/serverless-logic-install-kn-workflow-plugin-cli.adoc
+
+:_content-type: PROCEDURE
+[id="serverless-logic-install-kn-workflow-binary-file-linux_{context}"]
+= Installing the OpenShift Serverless Logic Knative Workflow plugin for Linux
```
[the remark this answers is not in the document]
Comment (adellape) [reply modules/serverless-logic-install-kn-workflow-binary-file-linux.adoc:7]: This peer-review request to switch to `{ServerlessProductName}` throughout the PR hasn't been applied yet.

**Review by adellape — COMMENTED** [review]
Holding merge briefly because we might as well switch to the common attribute as requested previously in https://github.com/openshift/openshift-docs/pull/80122/files#r1708774397.

File: modules/serverless-logic-install-kn-workflow-binary-file-linux.adoc [hunk 24]
```diff
@@ -0,0 +1,68 @@
...
+----
+
+. Rename the extracted Knative Workflow plugin binary file to `kn-workflow` by running the following command:
++
+[source,terminal]
+----
+$ mv <filename> kn-workflow
+----
+
+. Install the `kn-workflow` command as the Knative CLI plugin.
```
Comment (adellape) [inline modules/serverless-logic-install-kn-workflow-binary-file-linux.adoc:45]: IBM SG says use colons leading into substeps:
```suggestion
. Install the `kn-workflow` command as the Knative CLI plugin:
```

File: modules/serverless-logic-install-kn-workflow-binary-file-macos.adoc [hunk 25]
```diff
@@ -0,0 +1,61 @@
...
+----
+
+. Rename the extracted Knative Workflow plugin binary file to `kn-workflow` by running the following command:
++
+[source,terminal]
+----
+$ mv <filename> kn-workflow
+----
+
+. Install the `kn-workflow` command as the Knative CLI plugin.
```
Comment (adellape) [inline modules/serverless-logic-install-kn-workflow-binary-file-macos.adoc:38]:
```suggestion
. Install the `kn-workflow` command as the Knative CLI plugin:
```

File: modules/serverless-logic-install-kn-workflow-binary-file-windows.adoc [hunk 26]
```diff
@@ -0,0 +1,57 @@
...
+----
+
+. Rename the extracted Knative Workflow plugin binary file to `kn-workflow` by running the following command:
++
+[source,terminal]
+----
+$ Rename-Item -Path <destination>\<filename>.exe -NewName kn-workflow.exe
+----
+
+. Install the `kn-workflow` command as the Knative CLI plugin.
```
Comment (adellape) [inline modules/serverless-logic-install-kn-workflow-binary-file-windows.adoc:33]:
```suggestion
. Install the `kn-workflow` command as the Knative CLI plugin:
```

File: modules/serverless-logic-install-kn-workflow-binary-file-windows.adoc [hunk 27]
```diff
@@ -0,0 +1,57 @@
...
++
+On Windows, files like `exe` are treated as executable by default, so you do not need to change permissions.
```
Comment (adellape) [inline modules/serverless-logic-install-kn-workflow-binary-file-windows.adoc:35]: Dot to show file extensions:
```suggestion
On Windows, files like `.exe` are treated as executable by default, so you do not need to change permissions.
```
or maybe more clearly:
```suggestion
On Windows, files with a `.exe` extension are treated as executable by default, so you do not need to change permissions.
```

File: modules/serverless-logic-install-kn-workflow-binary-file-linux.adoc (hunk 23 above)
[the remark this answers is not in the document]
Comment (kaldesai) [reply modules/serverless-logic-install-kn-workflow-binary-file-linux.adoc:7]: I have updated the {ServerlessLogicProductName} attribute.

File: modules/serverless-logic-install-kn-workflow-binary-file-windows.adoc (hunk 22 above)
Comment (kaldesai) [reply modules/serverless-logic-install-kn-workflow-binary-file-windows.adoc]: Fixed!

File: modules/serverless-logic-install-kn-workflow-binary-file-linux.adoc (hunk 24 above)
Comment (kaldesai) [reply modules/serverless-logic-install-kn-workflow-binary-file-linux.adoc:45]: Updated wherever it is needed

**Comment by kaldesai** [comment]
@adellape, I have incorporated all the suggestions and the PR is ready now.

**Pull request merged**